substrate-node/runtime/src/lib.rs

```diff
@@ -508,7 +508,19 @@ impl ExtractRecipient for RecipientParser {
 	fn extract_recipient(dest: &MultiLocation) -> Option<Vec<u8>> {
 		// For example, we force a dest location should be represented by following format.
 		match (dest.parents, &dest.interior) {
-			(0, Junctions::X1(GeneralKey(recipient))) => Some(recipient.to_vec()),
+			(0, Junctions::X2(GeneralKey(recipient), GeneralIndex(_dest_domain_id))) =>
+				Some(recipient.to_vec()),
+			_ => None,
+		}
+	}
+}
+
+pub struct DestDomainIDParser;
+impl ExtractDestDomainID for DestDomainIDParser {
```

Comment:
I am wondering if we can merge `ExtractDestDomainID` and `ExtractRecipient` into one single trait, like `ExtractDest`, by returning (dest_domain_id, recipient) at the same time. So that developers don't need to implement two separate traits, I'd like to hear your opinion.

Reply:
I couldn't think of any reasons why not merging them at this moment, as they all have the same purpose: extracting something out from the same data. I will merge those.